Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertRebar.cs

```diff
@@ -1,175 +1,332 @@
-      var _closed = new List<DB.CurveArray>();
-
-      foreach (var curve in rebar.curves)
+      // skip freeform rebar for now: not supported by RevitRebarGroup class
+      // this is because freeform rebar with bent workshop has multiple shapes
+      if (revitRebar.GetAllRebarShapeIds().Count > 1)
       {
-        var array = CurveToNative(curve);
-        switch (curve)
-        {
-          case Polyline o:
-            if (o.closed)
-              _closed.Add(array);
-            else
-            {
-              var iterator = array.ForwardIterator();
-              while (iterator.MoveNext())
-                _open.Add(iterator.Current as DB.Curve);
-            }
-            break;
-          case Polycurve o:
-            if (o.closed)
-              _closed.Add(array);
-            else
-            {
-              var iterator = array.ForwardIterator();
-              while (iterator.MoveNext())
-                _open.Add(iterator.Current as DB.Curve);
-            }
-            break;
-          default:
-            _open.Add(array.get_Item(0));
-            break;
-        }
+        return null;
       }
 
-      open = _open; closed = _closed;
-    }
+      // get type
+      var type = revitRebar.Document.GetElement(revitRebar.GetTypeId()) as ElementType;
 
-    private Base RebarToSpeckle(DB.Structure.Rebar revitRebar)
-    {
-      // get rebar centerline curves using transform
-      var bars = revitRebar.GetCenterlineCurves(true, true, true, MultiplanarOption.IncludeOnlyPlanarCurves, revitRebar.NumberOfBarPositions - 1);
-      var curves = new List<ICurve>();
+      // get the rebar shape
+      var revitShape = revitRebar.Document.GetElement(revitRebar.GetShapeId()) as DB.Structure.RebarShape;
+      RevitRebarShape speckleShape = RebarShapeToSpeckle(revitShape);
+#if REVIT2020 || REVIT2021
+      speckleShape.barDiameter = revitRebar.GetBendData().BarDiameter;
+#else
+      speckleShape.barDiameter = revitRebar.GetBendData().BarModelDiameter;
+#endif
 
+      // get the rebar hooks
+      DB.ElementId revitStartHookId = revitRebar.GetHookTypeId(0);
+      RevitRebarHook speckleStartHook = null;
+      double hookBendRadius = revitRebar.GetBendData().HookBendRadius;
+      if (revitStartHookId != ElementId.InvalidElementId)
+      {
+        var revitStartHook = revitRebar.Document.GetElement(revitStartHookId) as RebarHookType;
+        speckleStartHook = RebarHookToSpeckle(
+          revitStartHook,
+          revitRebar.GetHookOrientation(0).ToString(),
+          hookBendRadius
+        );
+      }
+      DB.ElementId revitEndHookId = revitRebar.GetHookTypeId(1);
+      RevitRebarHook speckleEndHook = null;
+      if (revitEndHookId != ElementId.InvalidElementId)
+      {
+        var revitEndHook = revitRebar.Document.GetElement(revitEndHookId) as RebarHookType;
+        speckleEndHook = RebarHookToSpeckle(revitEndHook, revitRebar.GetHookOrientation(1).ToString(), hookBendRadius);
```

Comment:
Not sure I follow why we're turning this into a string

Reply:
The class property is a string, we could send the enum but this is not really human-readable after serialization